dali/operators/image/resize/tensor_resize_attr.cc

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/resize/tensor_resize_attr.h"
+#include <limits>
+#include <string>
+#include <vector>
+#include "dali/core/math_util.h"
+#include "dali/operators/image/resize/resize_attr.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+DALI_SCHEMA(TensorResizeAttr)
+    .AddOptionalArg<vector<float>>("sizes", R"code(Output sizes.
+
+When ``axes`` is provided, the size values refer to the axes specified.
+Note: Arguments ``sizes`` and ``scales`` are mutually exclusive.)code",
+                                   {}, true)
+    .AddOptionalArg<vector<float>>("scales", R"code(Scale factors.
+
+The resulting output size is calculated as
+``out_size = size_rounding(scale_factor * original_size)``.
+See ``size_rounding`` for a list of supported rounding policies.
+
+When ``axes`` is provided, the scale factor values refer to the axes specified.
+Note: Arguments ``sizes`` and ``scales`` are mutually exclusive.)code",
+                                   {}, true)
+    .AddOptionalArg<float>("alignment", R"code(Determines the position of the ROI
+when using scales (provided or calculated).
+
+The real output size must be integer and may differ from "ideal" output size calculated as input
+(or ROI) size multiplied by the scale factor. In that case, the output size is rounded (according
+to `size_rounding` policy) and the input ROI needs to be adjusted to maintain the scale factor.
+This parameter defines which relative point of the ROI should retain its position in the output.
+
+This point is calculated as ``center = (1 - alignment) * roi_start + alignment * roi_end``.
+Alignment 0.0 denotes alignment with the start of the ROI, 0.5 with the center of the region, and 1.0 with the end.
+Note that when ROI is not specified, roi_start=0 and roi_end=size is assumed.
+
+When using 0.5 (default), the resize operation has flip invariant properties (flipping after resizing is
+mathematically equivalent to resizing after flipping).
+
+Contains as many elements as dimensions provided for sizes/scales. If only one value is provided, it is
+applied to all dimensions.)code",
```

Comment:
Now, with multiple paragraphs above, it's hard to infer the subject.
```suggestion
The value of this argument contains as many elements as dimensions provided for sizes/scales.
If only one value is provided, it is applied to all dimensions.)code",
```